### ISSUE TITLE
[proposal][content] Integrate phpcr/phpcr Api

### ISSUE THREAD

**Issue by ndufreche** [backbee/BackBee#197]
The github repository [phpcr/phpcr](https://github.com/phpcr/phpcr) is a standardized api for the implementation of PHP Content Repository.

On the fact of BackBee try to implement the most standardized code possible, this Api can be great initiative and facilitate us the BackBee documentation for Content Repository.

**Comment by eric-chau** [comment]
I think it worth to take a look at this and think about a real implementation later when we are done with BBCoreJS first stable version.

**Comment by mickaelandrieu** [comment]
I'm interested too, because this will probably remove a lot of code of backbee/BackBee like nestednode or content. *But* is this as performant as our actual system ?

**Comment by ndufreche** [comment]
@mickaelandrieu this is not a php content repository implementation but just an API interface, we can keep your actual implementation.

Actually in the JS toolbar just partial part of the rest api is implemented.

**Comment by mickaelandrieu** [comment]
Not *remove* but extract our code to a ``backbee/phpcr`` repository maybe.

**Comment by ndufreche** [comment]
Yes it can be a good solution.

@mickaelandrieu there is lot of implementation of this Api like :
* [phpcr/phpcr-utils](https://github.com/phpcr/phpcr-utils)
* [jackalope/jackalope-doctrine-dbal](https://github.com/jackalope/jackalope-doctrine-dbal)
* [doctrine/DoctrinePHPCRBundle](https://github.com/doctrine/DoctrinePHPCRBundle)

**Comment by mickaelandrieu** [comment]
ping @lsmith77 there is multiple strategies:
* move our own implementation and (probably) implement all interfaces from phpcr;
* drop our own implementation and use doctrine dbal implementation;

We need to take a look to theses implementations, I prefer - if the dbal implementation is enough for our needs - use this one instead of maintain our own.

Then we can focus on more important tasks like UX, debugging, technical debt or new features :)

**Comment by lsmith77** [comment]
Sounds good .. I think the best course of action would be a 1-2 day event where we make this happen. @dbu is located in Bern and I live in Zurich. Like I said, I do plan to be at SymfonyCon in Paris in December ..

**Comment by mickaelandrieu** [comment]
@lsmith77 I'm pretty sure we can host you if you want :) let's talk about it when @crouillon will be available.

**Comment by dbu** [comment]
this sounds interesting. i will be in paris for symfony con as well. would be great to meet you guys then. best during the tutorial day or something, so that we can use the symfony con hackday for more open discussion with participants of the conference.

and having an own phpcr implementation without the phpcr interfaces does sound like technical debt to me - just saying ;-)

**Comment by mickaelandrieu** [comment]
yes, I will attend to the SymfonyCon (and you will as a speaker, congrats :-) ), agree that we need to respect the phpcr interfaces before doing any other improvements.
